### ISSUE TITLE
Single pixel instance in the upper left corner is missed by `kornia.contrib.connected_components`

### ISSUE THREAD

**Issue by Halyjo** [kornia/kornia#3169]
### Single pixel instance in the upper left corner is missed

The connected component algorithm used starts by placing unique values for every pixel in the foreground (with torch.arange) and zeroing out the background with a mask. The value of the first pixel is 0 (from the arange-function) by default which is confused for background in the spacial case above. The proposed solution is just to add one to all initial values.

### Reproduction steps

```bash
target = torch.tensor(
    [
        [1, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [1, 0, 0, 0, 0],
        [0, 0, 0, 0, 0],
        [1, 0, 1, 0, 1],
    ]
).view(1, 1, 5, 5)
out = connected_components(target)

-->

tensor([[[[ 0,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [10,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [22,  0, 24,  0, 26]]]])
```

### Expected behavior

-->
Maybe:
```
tensor([[[[ 1,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [11,  0,  0,  0,  0],
          [ 0,  0,  0,  0,  0],
          [23,  0, 25,  0, 27]]]])
```

### Environment

```shell
Not relevant.
```
